src/Exception/Exception.php

```diff
@@ -6,21 +6,44 @@
 
 class Exception extends \RuntimeException
 {
-    protected $errors = [];
+    private static $errorLabels = [
+        E_ERROR => 'E_ERROR',
+        E_WARNING => 'E_WARNING',
+        E_PARSE => 'E_PARSE',
+        E_NOTICE => 'E_NOTICE',
+        E_CORE_ERROR => 'E_CORE_ERROR',
+        E_CORE_WARNING => 'E_CORE_WARNING',
+        E_COMPILE_ERROR => 'E_COMPILE_ERROR',
+        E_COMPILE_WARNING => 'E_COMPILE_WARNING',
+        E_USER_ERROR => 'E_USER_ERROR',
+        E_USER_WARNING => 'E_USER_WARNING',
+        E_USER_NOTICE => 'E_USER_NOTICE',
+        E_STRICT => 'E_STRICT',
+        E_RECOVERABLE_ERROR => 'E_RECOVERABLE_ERROR',
+        E_DEPRECATED => 'E_DEPRECATED',
+        E_USER_DEPRECATED => 'E_USER_DEPRECATED',
+    ];
 
-    public function __construct($message, $code = 0, $previous = null)
+    final public function __construct($message, $code = 0, $previous = null)
```

Comment:
[Don't repeat yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself): with `final` keyword we ensure every Exception in this library follows this pattern